app/Http/Controllers/MaterialController.php

```diff
@@ -2,69 +2,82 @@
 
 namespace App\Http\Controllers;
 
+use App\Models\Employee;
 use App\Models\Material;
 use Illuminate\Http\Request;
-use Illuminate\Support\Facades\Route;
 
 class MaterialController extends Controller
 {
-    public function index()
+    const VALIDATION_RULES = [
+        'subject' => 'required',
+        'category' => 'required',
+        'text' => 'required',
+    ];
+
+    public function index(Request $request)
     {
-        return view('materials.index', [
-            'materials' => Material::query()->get()
+        $role = $request->get('role');
+        if (strlen($role) && in_array($role, Material::ROLES)) {
+            $materials = Material::query()->where('role', '=', $role)->get();
+        } else {
+            $materials = Material::query()->get();
+        }
```

Comment:
Тут вообще не будет никакой фильтрации. Поэтому вот так: 
```suggestion
        $materials = Material::query()->get();
```